reframe/core/modules.py

```diff
@@ -207,6 +208,30 @@ def is_module_loaded(self, name):
     def _is_module_loaded(self, name):
         return self._backend.is_module_loaded(Module(name))
 
+    def load_mapping_from_file(self, module_mapping_file):
+        """Update ``self.module_map`` with a mapping defined in a file."""
+        module_mapping = {}
+        try:
+            with open(module_mapping_file, 'r') as f:
+                for line in f.readlines():
+                    line = re.split(r'\s*#\s*', line.strip())[0]
```

Comment:
No need for regexes here. You can simply split the line at `#`. Everything at the left is your line of interest.

```python
line = line.split('#')[0].strip()
```